pyproject.toml

```diff
@@ -56,6 +56,7 @@ dev = [
     "sounddevice", # docs 
     "graphviz", #docs
     "pickleshare", #docs
+    "ruff>=0.3.5",
```

Comment:
I think it would be good to pin the version in order to ensure that CI is matching local ruff perfectly.
I would also suggest to use the latest version `0.4.1` which implements LSP.

Reply:
Indeed.